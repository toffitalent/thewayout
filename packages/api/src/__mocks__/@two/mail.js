const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
