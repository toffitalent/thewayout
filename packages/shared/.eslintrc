{
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": ["tsconfig.json", "packages/shared/tsconfig.json"]
      }
    }
  },
  "rules": {
    "no-param-reassign": "off",
    "react/react-in-jsx-scope": "off"
  }
}
