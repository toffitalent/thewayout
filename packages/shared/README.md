# @two/shared

Copyright (c) 2023-Present The Way Out, Inc. All Rights Reserved.

## Development

This package contains code shared between the [api](../api/README.md), [app](../app/README.md), and [web](../web/README.md) packages and is not intended to run in a standalone manner. See the READMEs of those packages for information on starting a development environment.

## Testing

To run all component and unit tests and get coverage reports, simply run `yarn workspace @dlatwobs/shared test`.
