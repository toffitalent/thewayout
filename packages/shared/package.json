{
  "name": "@two/shared",
  "version": "1.0.0",
  "author": "Disruptive Labs",
  "license": "UNLICENSED",
  "private": true,
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "scripts": {
    "build": "tsc -b tsconfig.build.json",
    "clean": "rimraf build coverage",
    "test": "jest --coverage",
    "type-check": "tsc"
  },
  "dependencies": {
    "@disruptive-labs/ui": "file:../../vendor/@disruptive-labs/ui",
    "axios": "^0.27.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/dom": "^9.3.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.2",
    "@types/react": "^18.2.14",
    "babel-jest": "^29.5.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "jest-junit": "^16.0.0",
    "react-test-renderer": "^18.2.0",
    "rimraf": "^3.0.2",
    "sass": "^1.63.6",
    "typescript": "^5.2.2"
  },
  "jest": {
    "clearMocks": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.{ts,tsx}"
    ],
    "globalSetup": "<rootDir>/test/setup.js",
    "moduleNameMapper": {
      "^.+\\.module\\.(css|sass|scss)$": "identity-obj-proxy",
      "^.+\\.(css|sass|scss)$": "<rootDir>/test/styleMock.js",
      "^.+\\.(jpg|jpeg|png|gif|webp|svg)$": "<rootDir>/test/fileMock.js",
      "^@test$": "<rootDir>/test/index.tsx"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/test/jest.test.ts"
    ],
    "testEnvironment": "jsdom",
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/test/"
    ],
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": [
        "babel-jest",
        {
          "presets": [
            "next/babel"
          ]
        }
      ]
    },
    "transformIgnorePatterns": [
      "node_modules/(?!(@?disruptive-labs.*|@popperjs/core|es-cookie)/)"
    ]
  },
  "jest-junit": {
    "outputDirectory": "./artifacts/jest",
    "outputName": "results.xml"
  }
}
