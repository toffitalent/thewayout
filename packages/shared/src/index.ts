// NOTE: React components should not exported here

export * from './constants';
export * from './types';
export * from './utils';
