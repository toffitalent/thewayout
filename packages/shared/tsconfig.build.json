{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "incremental": true,
    "module": "commonjs",
    "noEmit": false,
    "outDir": "./build",
    "rootDir": "./src",
    "tsBuildInfoFile": "./build/.tsbuildinfo"
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["build", "node_modules", "**/*.test.ts", "**/*.test.tsx", "test"]
}
