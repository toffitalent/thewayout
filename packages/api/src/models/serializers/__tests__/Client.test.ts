import { OffenseCategory, ReleasedAt, StateFederal, TimeServed } from '@two/shared';
import { fixtures, resetDb } from '@test';
import { Client, ClientSerializer } from '../..';

const getCtx = (admin?: boolean, self?: boolean) => ({
  auth: { is: () => !!self, has: () => !!admin },
});

describe('ClientSerializer', () => {
  beforeEach(async () => {
    await resetDb();
  });

  const offenseInformation = {
    offense: [OffenseCategory.arson],
    sexualOffenderRegistry: true,
    sbn: true,
    timeServed: TimeServed.timeServed01,
    releasedAt: ReleasedAt.releasedAt01,
    stateOrFederal: StateFederal.state,
  };

  test('only displays offense information to self or admins', async () => {
    const client = await Client.query()
      .insert({
        ...offenseInformation,
        userId: fixtures.user1.id,
        relativeExperience: [],
        education: [],
      })
      .returning('*');
    const serializer = new ClientSerializer();

    const serialized = await serializer.serialize(client);
    expect(serialized).toEqual(expect.not.objectContaining(offenseInformation));
    expect(serialized).toMatchSnapshot({ id: expect.any(String) });

    const serializedSelf = await serializer.serialize(client, { ctx: getCtx(false, true) });
    expect(serializedSelf).toEqual(expect.objectContaining(offenseInformation));
    expect(serializedSelf).toMatchSnapshot({ id: expect.any(String) });

    const serializedAdmin = await serializer.serialize(client, { ctx: getCtx(true, false) });
    expect(serializedAdmin).toEqual(expect.objectContaining(offenseInformation));
    expect(serializedAdmin).toMatchSnapshot({ id: expect.any(String) });
  });
});
