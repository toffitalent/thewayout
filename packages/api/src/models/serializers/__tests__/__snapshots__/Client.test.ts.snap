// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientSerializer only displays offense information to self or admins 1`] = `
{
  "address": null,
  "age": null,
  "city": null,
  "disability": null,
  "education": [],
  "ethnicity": null,
  "experience": null,
  "firstName": null,
  "gender": null,
  "id": Any<String>,
  "justiceStatus": null,
  "languages": null,
  "lastName": null,
  "license": null,
  "maritalStatus": null,
  "orientation": null,
  "personalStrengths": null,
  "personalSummary": null,
  "phone": null,
  "postalCode": null,
  "referredBy": null,
  "relativeExperience": [],
  "religion": null,
  "state": null,
  "support": null,
  "updatedAt": "<Date>",
  "userId": "01817e24-0d42-7176-b613-889fe94905f0",
  "veteranOrJustice": [
    "justiceImpacted",
  ],
  "veteranStatus": null,
}
`;

exports[`ClientSerializer only displays offense information to self or admins 2`] = `
{
  "address": null,
  "age": null,
  "city": null,
  "disability": null,
  "education": [],
  "ethnicity": null,
  "expectedReleasedAt": null,
  "experience": null,
  "facility": null,
  "firstName": null,
  "gender": null,
  "id": Any<String>,
  "justiceStatus": null,
  "languages": null,
  "lastName": null,
  "license": null,
  "maritalStatus": null,
  "offense": [
    "arson",
  ],
  "orientation": null,
  "personalStrengths": null,
  "personalSummary": null,
  "phone": null,
  "postalCode": null,
  "referredBy": null,
  "relativeExperience": [],
  "releasedAt": "releasedAt01",
  "releasedCounty": null,
  "religion": null,
  "sbn": true,
  "sexualOffenderRegistry": true,
  "state": null,
  "stateOrFederal": "state",
  "support": null,
  "timeServed": "timeServed01",
  "updatedAt": "<Date>",
  "userId": "01817e24-0d42-7176-b613-889fe94905f0",
  "veteranOrJustice": [
    "justiceImpacted",
  ],
  "veteranStatus": null,
}
`;

exports[`ClientSerializer only displays offense information to self or admins 3`] = `
{
  "address": null,
  "age": null,
  "city": null,
  "disability": null,
  "education": [],
  "ethnicity": null,
  "expectedReleasedAt": null,
  "experience": null,
  "facility": null,
  "firstName": null,
  "gender": null,
  "id": Any<String>,
  "justiceStatus": null,
  "languages": null,
  "lastName": null,
  "license": null,
  "maritalStatus": null,
  "offense": [
    "arson",
  ],
  "orientation": null,
  "personalStrengths": null,
  "personalSummary": null,
  "phone": null,
  "postalCode": null,
  "referredBy": null,
  "relativeExperience": [],
  "releasedAt": "releasedAt01",
  "releasedCounty": null,
  "religion": null,
  "sbn": true,
  "sexualOffenderRegistry": true,
  "state": null,
  "stateOrFederal": "state",
  "support": null,
  "timeServed": "timeServed01",
  "updatedAt": "<Date>",
  "userId": "01817e24-0d42-7176-b613-889fe94905f0",
  "veteranOrJustice": [
    "justiceImpacted",
  ],
  "veteranStatus": null,
}
`;
